controllers/apply-next/lib/pagination.js

```diff
@@ -37,14 +32,14 @@ function findNextMatchingUrl({ baseUrl, sections, currentSectionIndex, currentSt
  * Find previous matching URL
  * @param {MatchOptions} options
  */
-function findPreviousMatchingUrl({ baseUrl, sections, currentSectionIndex, currentStepIndex, formData }) {
+function findPreviousMatchingUrl({ baseUrl, sections, currentSectionIndex, currentStepIndex }) {
     const currentSection = sections[currentSectionIndex];
     const previousSection = sections[currentSectionIndex - 1];
 
     if (currentStepIndex !== 0) {
         const targetStepIndex = findLastIndex(
             currentSection.steps,
-            step => (step.matchesCondition ? step.matchesCondition(formData) === true : true),
```

Comment:
Avoids duplicating matching logic here and directly checks the new `step.isRequired` property.